src/main/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParser.java

```diff
@@ -1,29 +1,75 @@
 package co.rsk.bitcoinj.script;
 
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Sha256Hash;
 import co.rsk.bitcoinj.core.Utils;
 import co.rsk.bitcoinj.core.VerificationException;
+import java.util.Collections;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class P2shErpFederationRedeemScriptParser extends StandardRedeemScriptParser {
+public class P2shErpFederationRedeemScriptParser implements RedeemScriptParser {
     private static final Logger logger = LoggerFactory.getLogger(P2shErpFederationRedeemScriptParser.class);
 
     public static long MAX_CSV_VALUE = 65_535L; // 2^16 - 1, since bitcoin will interpret up to 16 bits as the CSV value
 
+    private final MultiSigType multiSigType;
+    private final RedeemScriptParser redeemScriptParser;
+    private final ScriptType scriptType;
+
     public P2shErpFederationRedeemScriptParser(
         ScriptType scriptType,
         List<ScriptChunk> redeemScriptChunks,
         List<ScriptChunk> rawChunks
     ) {
-        super(
-            scriptType,
-            extractStandardRedeemScriptChunks(redeemScriptChunks),
-            rawChunks
-        );
+        this.scriptType = scriptType;
+        List<ScriptChunk> standardRedeemScriptChunks = extractStandardRedeemScriptChunks(redeemScriptChunks);
+        this.redeemScriptParser = new StandardRedeemScriptParser(scriptType, standardRedeemScriptChunks, rawChunks);
         this.multiSigType = MultiSigType.P2SH_ERP_FED;
+
+    }
+
+    @Override
+    public MultiSigType getMultiSigType() {
+        return multiSigType;
+    }
+
+    @Override
+    public ScriptType getScriptType() {
+        return scriptType;
+    }
+
+    @Override
+    public int getM() {
+        return this.redeemScriptParser.getM();
+    }
+
+    @Override
+    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
+        return this.redeemScriptParser.getSigInsertionIndex(hash, signingKey);
+    }
+
+    @Override
+    public int findKeyInRedeem(BtcECKey key) {
+        return this.redeemScriptParser.findKeyInRedeem(key);
+    }
+
+    @Override
+    public List<BtcECKey> getPubKeys() {
+        return this.redeemScriptParser.getPubKeys();
+    }
+
+    @Override
+    public int findSigInRedeem(byte[] signatureBytes, Sha256Hash hash) {
+        return this.redeemScriptParser.findSigInRedeem(signatureBytes, hash);
+    }
+
+    @Override
+    public List<ScriptChunk> extractStandardRedeemScriptChunks() {
+        return this.redeemScriptParser.extractStandardRedeemScriptChunks();
     }
 
     public static List<ScriptChunk> extractStandardRedeemScriptChunks(List<ScriptChunk> chunks) {
```

Comment:
What do you think about moving this method implementation to `extractStandardRedeemScriptChunks` that overrides the interface contract? After that, we might get rid of this `static` method.

Reply:
We need to analyze this change in detail. This is not part of the scope of this task. Let's follow up on this proposal in the analysis task.

Reply:
Agree. We can create a new task to do it.